src/protections/TrustedReporters.ts

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright 2022 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import config from "../config";
+import { Protection } from "./IProtection";
+import { MXIDListProtectionSetting, NumberProtectionSetting, OptionListProtectionSetting } from "./ProtectionSettings";
+import { Mjolnir } from "../Mjolnir";
+
+export class TrustedReporters extends Protection {
+    private recentReported = new Map<string, Set<string>>();
+
+    settings = {
+        "mxids": new MXIDListProtectionSetting(),
+        "threshold": new NumberProtectionSetting(3),
+        "action": new OptionListProtectionSetting(
+            "alert",
+            ["alert", "redact", "ban"]
+        )
+    };
+
+    constructor() {
+        super();
+    }
+
+    public get name(): string {
+        return 'TrustedReporters';
+    }
+    public get description(): string {
+        return "Count reports from trusted reporters and take a configured action";
+    }
+
+    public async handleReport(mjolnir: Mjolnir, roomId: string, reporterId: string, reason: string, event: any): Promise<any> {
```

Comment:
At this stage, it would be really nice to have an idea of what you're expecting in `event` :)

Reply:
I'm unsure what `event`s are, given `matrix-bot-sdk` also seems to not do something concrete for them: https://github.com/turt2live/matrix-bot-sdk/blob/e0094141f6403d790a5f0c2c6d29b9e779971c41/src/MatrixClient.ts#L856

Reply:
Let's try and do better than matrix-bot-sdk, if we can :)

But seriously, is it an entire event? An event `content`? Are you expecting any specific properties?

Reply:
I think that I'm being over-pedantic. Please ignore me :)

Reply:
I think you're right that we should be making this more strongly typed, but i think it might be a bigger piece of work